python/lsst/pipe/tasks/processBrightStars.py

```diff
@@ -0,0 +1,383 @@
+        det = stamps[0].getDetector()
+        # Define correction for optical distortions
+        pixToTan = det.getTransform(cg.PIXELS, cg.TAN_PIXELS)
+        # Array of all possible rotations for detector orientation:
+        possibleRots = np.array([k*np.pi/2 for k in range(4)])
+        # determine how many, if any, rotations are required
+        yaw = det.getOrientation().getYaw()
+        nb90Rots = np.argmin(np.abs(possibleRots - float(yaw)))
+
+        # apply transformation to each star
+        warpedStars = []
+        for star, cent in zip(stamps, pixCenters):
+            # (re)create empty destination image
+            destImage = afwImage.MaskedImageF(*self.modelStampSize)
+            bottomLeft = geom.Point2D(star.getImage().getXY0())
+            newBottomLeft = pixToTan.applyForward(bottomLeft)
+            newBottomLeft.setX(newBottomLeft.getX() - bufferPix[0]/2)
+            newBottomLeft.setY(newBottomLeft.getY() - bufferPix[1]/2)
+            # Convert to int
+            newBottomLeft = geom.Point2I(newBottomLeft)
+            # Set origin
+            destImage.setXY0(newBottomLeft)
+
+            # Define linear shifting to recenter stamps
+            newCenter = pixToTan.applyForward(cent)  # center of warped star
+            shift = self.modelCenter[0] + newBottomLeft[0] - newCenter[0],\
+                self.modelCenter[1] + newBottomLeft[1] - newCenter[1]
+            affineShift = geom.AffineTransform(shift)
+            shiftTransform = tFactory.makeTransform(affineShift)
+
+            # Define full transform (warp and shift)
+            starWarper = pixToTan.then(shiftTransform)
+
+            # Apply it
+            goodPix = afwMath.warpImage(destImage, star.getMaskedImage(),
+                                        starWarper, warpCont)
+            if not goodPix:
+                self.log.debug("Warping of a star failed: no good pixel in output")
+
+            # Arbitrarily set origin of shifted star to 0
+            destImage.setXY0(0, 0)
+
+            # Apply rotation if apropriate
+            if nb90Rots:
+                destImage = afwMath.rotateImageBy90(destImage, nb90Rots)
+            warpedStars.append(destImage.clone())
+        return warpedStars
+
+    @pipeBase.timeMethod
+    def run(self, inputExposure, refObjLoader=None, dataId=None):
+        """Identify bright stars within an exposure using a reference catalog,
+        extract stamps around each, then preprocess them. The preprocessing
+        steps are: shifting, warping and potentially rotating them to the same
+        pixel grid; computing their annular flux and normalizing them.
+
+        Parameters
+        ----------
+        inputExposure : `afwImage.exposure.exposure.ExposureF`
+            The image from which bright star stamps should be extracted.
+        refObjLoader : `LoadIndexedReferenceObjectsTask`, optional
+            Loader to find objects within a reference catalog.
+        dataId : `dict`
```

Comment:
In gen3 world, this will be a lsst.daf.butler.DataId (which itself is already either a DataCoordinate or dict). It should be included in the documentation, and it is up to you if you want to also include dict (as one might not know DataId could be a dict) or you could document it as "`dict` or `lsst.daf.butler.DataCoordinate`"